my_game.py

```diff
@@ -481,6 +481,24 @@ def get_explosion(self, position, textures=None):
             particle_lifetime_max=CONFIG['EXPLOSION_PARTICLE_LIFETIME_MAX'],
             particle_scale=CONFIG['EXPLOSION_PARTICLE_SIZE'])
 
+    def shockwave(self, center: tuple[float, float], range: float, strength: float, sprites: arcade.SpriteList):
+        """
+        create a shockwave at the center that pushes away all given sprites
+        """
+
+        for sprite in sprites:
+            dist = arcade.get_distance(sprite.center_x, sprite.center_y, center[0], center[1])
+
+            if dist <= range:
+
+                # point away from the center
+                sprite.angle = arcade.get_angle_degrees(sprite.center_x, sprite.center_y, center[0], center[1]) - 180
+
+                # the closer to the center, the faster it moves
+                impact = abs(dist / range - 1) * strength
+                sprite.change_x += math.sin(sprite.radians) * impact
+                sprite.change_y += math.cos(sprite.radians) * impact
```

Comment:
Let's use _sprite.forward()_ here like we do elsewhere.

Reply:
well, there's a problem: arcade.forward uses cos on x and sin on y, and in my code its reversed...
I assume arcades code is more mathematically correct, but it doesnt work with that code.